### PR TITLE
#23: Adding roll me command for dice-rolling support

### LINKED ISSUE #23: Roll Command
Rolls one or more dice. Defaults to 1d6, max is 300d20.

### PR DESCRIPTION
**Pull request by jedibob5**
Resolves #23 

Dice-rolling command. Pretty straightforward. Syntax is `roll me [number of dice]d[number of sides]`, i.e. `roll me 3d8`. Includes a alias to just `roll` for convenience's sake.

The feature request (#23) specified a maximum of 300d20, but d100s are fairly common in TTRPGs, plus the die size doesn't really have any performance impact, so I have it at a max of 300d1000 right now. I'm open to suggestions if this should be changed, though I tested it on the highest possible roll and the response time is still pretty quick.

Also fixed a couple of minor typos while I was at it, and added a little Easter egg, if that's acceptable. Don't tell anyone about it right away, I'd like to see if anyone stumbles onto it first.

### CODE REVIEW

File: lisa_commands/commands.py [hunk 1]
```diff
@@ -100,6 +100,22 @@ def commands():
...
         },
+        'roll me': {
+            'function': roll_me,
+            "description": """
+            Roll the dice! Usage:
+            `roll me`: defaults to rolling 1d6
+            `roll me 3d6`: rolls three d6es
+            `roll me 4d20`: rolls four d20s :snoop:
+
+            Maximum die size: d100
```
Comment (jedibob5) [inline lisa_commands/commands.py:111]: ope, that should be 1000, will fix after work

File: lisa_commands/commands.py [hunk 2]
```diff
@@ -459,14 +475,57 @@ def reverse_me(query, _slack_event):
...
+    """
+    rolls the dice. maximum die size: d100, maximum number of dice: 300
+    :param query: query string
+    :param _slack_event: a dict of slack event information (unused for this function)
+    :return: a string with the result
+    """
+    # lisa can have little a easter egg, as a treat.
+    # video is the smoke weed every day song
+    if query == "a joint" or query == "a blunt":
+        return ":mokeweed::snoop: https://www.youtube.com/watch?v=JcBIzuuIdOA :mokeweed::snoop:"
```
Comment (Splyth) [inline lisa_commands/commands.py:488]: I like it 😀

File: lisa_commands/commands.py [hunk 3]
```diff
@@ -459,14 +475,57 @@ def reverse_me(query, _slack_event):
     """
     return query[::-1]
 
+def roll_me(query, _slack_event):
+    """
+    rolls the dice. maximum die size: d100, maximum number of dice: 300
```
Comment (Splyth) [inline lisa_commands/commands.py:480]: Should be 1000 right?

Comment (jedibob5) [reply lisa_commands/commands.py]: Fixed

**Comment by jedibob5** [comment]
@Zylox @noah-guillory @Mill1893 Need another review on this when yall have some time

File: lisa_commands/commands.py [hunk 4]
```diff
@@ -459,14 +475,57 @@ def reverse_me(query, _slack_event):
...
+    # lisa can have little a easter egg, as a treat.
+    # video is the smoke weed every day song
+    if query == "a joint" or query == "a blunt":
+        return ":mokeweed::snoop: https://www.youtube.com/watch?v=JcBIzuuIdOA :mokeweed::snoop:"
+
+    if query == "":
+        roll = [1, 6]
+    else:
+        roll = query.split('d', 1)
+    if len(roll) != 2:
```
Comment (amiller-sta) [inline lisa_commands/commands.py:494]: I think it'd be nice if we could default to 1 when the user doesn't supply a number, but it's not a big deal and might not be a use case people other than me really think of.

**Pull request merged**